### ISSUE TITLE
Numeric key triggers colon rule

### ISSUE THREAD

**Issue by msalmonse** [realm/SwiftLint#2839]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The colon rule is triggered if a dictionary has a numeric key

### Environment

* SwiftLint version: 0.34.0
* Installation method used: Downloaded package
* Paste your configuration file:

```yml
disabled_rules:
  - comma
  - identifier_name
  - private_over_fileprivate
  - type_name
```

* Are you using nested configurations No
* Which Xcode version are you using 11.0 Beta 5
* Do you have a sample that shows the issue? 
```swift
// This triggers a violation:
       let testDict: [Int: [Int]] = [
            6:   [ 6, 22, 38, 54, 70, 86, 102, 118, 134, 150, 166, 182, 198, 214, 230, 246 ],
            223: [ 15, 31, 47, 63, 79, 95, 111, 127, 143, 159, 175, 191, 207, 223, 239, 255 ]
        ]
```

**Comment by marcelofabri** [comment]
This is working as expected, you need to use a configuration if you want to have multiple spaces on the right side:

```yml
disabled_rules:
  - comma
  - ...

colon:
  flexible_right_spacing: true
```
